File: GLLara/GLLAppDelegate.m
//
//  GLLAppDelegate.m
//  GLLara
//
//  Created by Torsten Kammer on 01.12.15.
//  Copyright (c) 2015 Torsten Kammer. All rights reserved.
//

#import "GLLAppDelegate.h"

#import "GLLPreferenceKeys.h"
#import "GLLPreferencesWindowController.h"
#import "GLLara-Swift.h"

@interface GLLAppDelegate() {
    GLLHIDManager *manager;
}

@end

@implementation GLLAppDelegate

- (void)awakeFromNib
{
    [[NSUserDefaults standardUserDefaults] registerDefaults:@{
        GLLPrefUseAnisotropy: @(YES),
        GLLPrefUseMSAA: @(YES),
        GLLPrefAnisotropyAmount: @(4),
        GLLPrefMSAAAmount: @(4),
        GLLPrefObjExportIncludesTransforms: @(YES),
        GLLPrefObjExportIncludesVertexColors: @(NO),
        GLLPrefPoseExportIncludesUnused: @(NO),
        GLLPrefPoseExportOnlySelected: @(YES),
        GLLPrefShowSkeleton: @(YES)
    }];
    
    manager = [[GLLHIDManager alloc] init];
}

- (IBAction)openPreferences:(id)sender;
{
    if (!self.preferencesWindowController) {
        self.preferencesWindowController = [[GLLPreferencesWindowController alloc] init];
    }
    
    [self.preferencesWindowController showWindow:sender];
}

@end

File: GLLara/GLLAppDelegate.swift
//
//  GLLAppDelegate.swift
//  GLLara
//
//  Created by Torsten Kammer on 09.07.22.
//  Copyright © 2022 Torsten Kammer. All rights reserved.
//

import Cocoa

@objc class GLLAppDelegate: NSObject, NSApplicationDelegate {
    
    var preferencesWindowController: GLLPreferencesWindowController? = nil
    
    override init() {
        UserDefaults.standard.register(defaults: [
            GLLPrefUseAnisotropy: true,
            GLLPrefUseMSAA: false,
            GLLPrefAnisotropyAmount: 4,
            GLLPrefMSAAAmount: 2,
            GLLPrefObjExportIncludesTransforms: true,
            GLLPrefObjExportIncludesVertexColors: false,
            GLLPrefPoseExportIncludesUnused: false,
            GLLPrefPoseExportOnlySelected: true,
            GLLPrefShowSkeleton: true,
            GLLPrefHideUnusedBones: true,
            GLLPrefSpaceMouseSpeedTranslation: 1,
            GLLPrefSpaceMouseDeadzoneTranslation: 0.0,
            GLLPrefSpaceMouseSpeedRotation: 90.0 * Double.pi / 180.0,
            GLLPrefSpaceMouseDeadzoneRotation: 0.0,
            GLLPrefSpaceMouseMode: GLLView.CameraMovementMode.rotateAroundTarget.rawValue,
            GLLPrefControllerCameraRotationSpeed: 30.0 * Double.pi / 180.0,
            GLLPrefControllerCameraMovementSpeed: 1.0,
            GLLPrefControllerBoneMovementSpeed: 0.02,
            GLLPrefControllerBoneRotationSpeed: 11.25 * Double.pi / 180.0
        ])
    }
    
    @IBAction func openPreferences(sender: Any?) {
        if (preferencesWindowController == nil) {
            preferencesWindowController = GLLPreferencesWindowController()
        }
        
        preferencesWindowController!.showWindow(sender)
    }
}
